src/explorer/Explorers/NumericColumnExplorer.cs

```diff
@@ -0,0 +1,28 @@
+namespace Explorer.Explorers
+{
+    using Diffix;
+    using Explorer.Common;
+    using Explorer.Explorers.Components;
+
+    internal class NumericColumnExplorer : ExplorerBase
+    {
+        public NumericColumnExplorer(DConnection conn, ExplorerContext ctx)
+        : base(conn, ctx)
+        {
+        }
+
+        protected override void InitializeComponents()
+        {
+            Initialize<SimpleStats<double>, SimpleStats<double>.Result>()
+                .LinkToDependentComponent((NumericHistogramComponent)
+                    Initialize<NumericHistogramComponent, NumericHistogramComponent.Result>()
+                    .LinkToDependentComponent(
+                        Initialize<QuartileEstimator, QuartileEstimator.Result>())
+                    .LinkToDependentComponent(
+                        Initialize<AverageEstimator, AverageEstimator.Result>()));
```

Comment:
I guess you mean that with a list of components, it should be possible to connect them as required? I suppose maybe there's a way to wire everything up automatically using some reflection magic (or dependency injection 😱) but it sounds tricky. It's also nice to have an explicit view of the dependencies.

Reply:
Yes, I thought about using reflection and something similar to how dependency injection works in ASP.NET core, i.e. we could make the components that are dependencies to be parameters to the Explore method. (I imagine that ASP.NET core dependency injection is also using reflection  internally.)
Regrading the explicit view of the dependencies, the following code seems pretty explicit to me in the way it express the dependency:

`internal class NumericHistogramComponent :
        ExplorerComponent<NumericHistogramComponent.Result>,
        DependsOn<SimpleStats<double>.Result>`.

I still do not understand why we need to specify the same thing in a second place:

`            Initialize<SimpleStats<double>, SimpleStats<double>.Result>()
                .LinkToDependentComponent((NumericHistogramComponent)
                    Initialize<NumericHistogramComponent, NumericHistogramComponent.Result>()`

If necessary we could also add type of the component as an additional type parameter to the DependsOn interface in addition to the result type.

Reply:
There is one more thing: the piece of code in `NumericColumnExplorer.InitializeComponents` is very difficult to read for me.

Reply:
> I think that we could have an Initialize method that will walk the chain of DependsOn and create all the necessary components. And then using some more reflection magic, we can pass them to the Explore method.

This is a nice idea, I'll look into it. However I'm worried that this will involve implementing dependency injection, in which case we might as well use the DI already included with ASP.NET Core. Then we would need to pass the dependencies in the constructor and could remove the `DependsOn` interface. Maybe this is a better option? 

Also I'm not at all familiar with the reflection api and don't want to sink too much time into this. 

Either way, it would be good to get this right, it might have benefits for testing too (we can test individual components instead of the whole explorer, we can provide mock components that provide a result without hitting the database etc.).

>There is one more thing: the piece of code in NumericColumnExplorer.InitializeComponents is very difficult to read for me.

I agree with this but I struggled to find a better way to express it.